Expanded the README to state some contribution rules (#13)
singularityhub/shpc-registry: README.md

📖️ Read the [documentation](https://singularity-hpc.readthedocs.io/en/latest/) 📖️
⭐️ Browse the [container module collection](https://singularityhub.github.io/shpc-registry/) ⭐️

> ## Contribution rules

Reviewer: ```suggestion
## Contribution
```
They are technically rules, but I think just "Contribution" gets the message across!

> This registry is there to showcase Singularity HPC, and provide a workable default set of containers.

Reviewer: ```suggestion
This registry showcases Singularity HPC (shpc), and provides the default set of containers for its default registry.
```

> Contributions are very much welcome, so please do submit a pull-request if you'd like more software to be added !

Reviewer: yay!

> The documentation has an [introduction](https://singularity-hpc.readthedocs.io/en/latest/getting_started/developer-guide.html#writing-registry-entries) about registries, and explain how to organise the various files.

Reviewer: ```suggestion
The documentation has an [introduction](https://singularity-hpc.readthedocs.io/en/latest/getting_started/developer-guide.html#writing-registry-entries) about registries, and explains how to organize the required files for a container entry.
```

### Automated generation

First of all, there is an automated way of getting most of the `container.yaml` written via GitHub magic 🎩 automation.

> 1. [Fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) of this repository.

Reviewer: Do they need to fork the repository? E.g., should we make an issue template that requests a container, and then we can easily trigger (I'm happy to do this). So maybe we could suggest to open an issue to request a container URI, and (optionally) for advanced users they can fork, run the workflow, and PR.

Author: How would you handle editing the list of aliases ? By default, people wouldn't be allowed to edit the PR. Unless you change the permissions for that branch ?

Reviewer: Hmm that's a good point - do you want to try it out and see what we can do?

Author: I've added an issue form 7692c0dcf3214f3adda6fe34aa23d9d5aec872e7 to collect the requirements

2. Go to the Actions tab.
3. Select "Generate New Container" in the left-hand side menu.
4. Click the "Run workflow" dropdown on the right-hand side.
5. Fill in the name of the container you want to build a `container.yaml` for, as well as a URL and a description (both are mandatory fields). Leave the branch as `main`.

> 6. Hit the green "Run workflow" button, wait and marvel at the magic happening. The GitHub Action will automatically open a pull request with most of the `container.yaml` filled in for you. You can check the branch out, edit it further – in particular refine the list aliases.

Reviewer: ```suggestion
6. Hit the green "Run workflow" button, wait and marvel at the magic :sparkles: happening. The GitHub Action will automatically open a pull request with most of the `container.yaml` filled in for you. You can check the branch out, edit it further – in particular refine the list aliases.
```

### Expected content

Refer to the documentation for a list of the [`container.yaml` fields](https://singularity-hpc.readthedocs.io/en/latest/getting_started/developer-guide.html#registry-yaml-fields), especially the _required_ ones.

> A pull-request that doesn't contain the required fields unfortunately 😢 cannot be accepted.

Reviewer: ```suggestion
A pull-request that doesn't contain the required fields will have changes requested to add them.
```

📖️ Read the [documentation](https://singularity-hpc.readthedocs.io/en/latest/) 📖️
⭐️ Browse the [container module collection](https://singularityhub.github.io/shpc-registry/) ⭐️

## Contribution

This registry showcases Singularity HPC (shpc), and provides the default set of containers for its default registry.

> Contributions are very much welcome, so please do submit a pull-request if you'd like more software to be added !

Reviewer: ```suggestion
Contributions are very much welcome, so please do submit a pull-request if you'd like more software to be added, or open an issue to request a new addition!
```

Reviewer: I think that's the last little tweak so they are aware of the issue, then we are good for  merge!
